### ISSUE TITLE
Use BlueBubbles behind lighttpd reverse proxy beneath a subdir instead of an alternate host?

### ISSUE THREAD

**Issue by jxyzn** [BlueBubblesApp/bluebubbles-server#611]
I'd like to put BlueBubbles on my host with an existing SSL cert.

With lighttpd, I thought I could get this working with a lighttpd config section like this on "host.my.domain":

```
server.modules   += ( "mod_proxy" )
$HTTP["url"] =~ "^/bluebubbles/" {
   # WebSocket proxy
   proxy.server  = ( "" => ("internal.mac.ip.addr" => ( "host" => "internal.mac.ip.addr", "port" => 1234 )))
   proxy.replace-http-host = "enable"
   proxy.forwarded = ( "for" => 1 )
   proxy.header = (
      "https-remap" => "enable",
      "map-urlpath" => (
          "/bluebubbles" => "",
      ),
   )
}
```

This would let SSL-secured access to **https://host.my.domain:443/bluebubbles/*** be forwarded (minus the SSL) to **http://internal.mac.ip.addr:1234/** (with the same path info from the original URL up to "bluebubbles/" removed).

Setting my server URL to "**https://host.my.domain/bluebubbles/**" in the web app does not work: In trying this, I see the web app fails due to attempting to access **https://host.my.domain/api/v1/...etc...**, ignoring the path I specified. Additionally, changing the config to catch "**/api/v1/**" and pass those along, it fails with **/socket.io/** path URLs a little later. And adding that path to the patterns forwarded still has some websocket errors.

Is supporting a sub-path on the URL something that could be supported, or is there a structural problem with supporting that?

Thanks

**Comment by jxyzn** [comment]
I've generalized the description a bit beyond my specific use-case. Given that an entire URL is allowed for the server URL, I assumed you could specify a top-level path under which the service could be accessed, but it appears the server ignores any path after the port number.

**Comment by zlshames** [comment]
Would I be correct to say this is a client-side bug? Like the client isn't honoring your full path and instead just using the hostname. 

If so, I can transfer the issue to the client

**Comment by jxyzn** [comment]
Hm. Not sure. I suppose that could be it. I'm seeing similar connection problems on both Android and the web client, and that seems perhaps likely based on what the status reported in the Android client shows.

If I configure my server to hand out URL  "https://example.com/bb" and then tell the web client to use that URL, I get:

"Failed to connect to https://qixl.jx.io/bb!"

And the Network panel in the Chrome Inspector shows 404 errors attempting to retrieve  two pages:

https://example.com/api/v1/server/info?guid=my_password
https://example.com/api/v1/fcm/client?guid=my_password

So it's not present there.

So the web client knows the correct URL, at least to report the failure.


If I configure the Android client with that URL, the Connection & Server details page shows:

API Connection: CONNECTING
Socket Connection: CONNECTING/ERROR/DISCONNECTED (repeating)
Server URL: https://example.com

The /bb is not shown. <- this does seem suspicious on the client.

**Comment by zlshames** [comment]
> Hm. Not sure. I suppose that could be it. I'm seeing similar connection problems on both Android and the web client, and that seems perhaps likely based on what the status reported in the Android client shows.
> 
> If I configure my server to hand out URL "https://example.com/bb" and then tell the web client to use that URL, I get:
> 
> "Failed to connect to https://qixl.jx.io/bb!"
> 
> And the Network panel in the Chrome Inspector shows 404 errors attempting to retrieve two pages:
> 
> https://example.com/api/v1/server/info?guid=my_password https://example.com/api/v1/fcm/client?guid=my_password
> 
> So it's not present there.
> 
> So the web client knows the correct URL, at least to report the failure.
> 
> If I configure the Android client with that URL, the Connection & Server details page shows:
> 
> API Connection: CONNECTING Socket Connection: CONNECTING/ERROR/DISCONNECTED (repeating) Server URL: https://example.com
> 
> The /bb is not shown. <- this does seem suspicious on the client.

yeah my guess is that the client is just taking the hostname of your URL, rather than appending the API route to the base URL. So your Server URL is really just `<schema>://<hostname>`, and any URL path is ignored. I'll move this to the App's repo since I think this is just the way the client is right now, and we can/should change it for this use-case
